### PR TITLE
React library management system

### PR DESCRIPTION
**Pull request by SuduguTejasvi**

### CODE REVIEW

**Comment by SuduguTejasvi** [comment]
image

**Review by MajaharZemoso — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+import UpdateBook from "./components/organisms/UpdateBooks";
+import NotFound from "./components/organisms/NotFoundPage";
+import { ThemeProvider } from "@mui/material";
+import { Theme } from "./themes";
+import BookStorePage from "./pages";
+import StudentInformation from "./components/StudentInformation";
+const App: React.FC = () => {
+    const students=[{id:'1',name:'tejasvi',marks:220},{id:'2',name:'priya',marks:200},{id:'3',name:'riya',marks:260}]
+    return (
+        <>
```
Comment (MajaharZemoso) [inline src/App.tsx:16]: Don't use empty fragments

File: src/components/StudentInformation.tsx
Comment (MajaharZemoso) [inline src/components/StudentInformation.tsx:1]: Delete the file if it is not related to this PR

File: src/components/molecules/Actions/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { IconButton, Box } from '@mui/material';
+import { Delete, Edit, Visibility } from '@mui/icons-material';
+import { dataactionsprops } from '../../../utils/interfaces';
+
+type IconButtonColor = 'inherit' | 'primary' | 'secondary' | 'default' | 'error' | 'info' | 'success' | 'warning';
```
Comment (MajaharZemoso) [inline src/components/molecules/Actions/index.tsx:6]: add types also in interface file

File: src/components/molecules/Actions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const actions = [
+    { action: handleGetData, color: 'primary' as IconButtonColor, label: 'get data', icon: <Visibility /> },
+    { action: handleUpdateData, color: 'secondary' as IconButtonColor, label: 'update data', icon: <Edit /> },
+    { action: handleDeleteData, color: 'error' as IconButtonColor, label: 'delete data', icon: <Delete /> },
+  ];
+
+  return (
+    <Box>
+      {actions.map(({ action, color, label, icon }, index) => (
+        <IconButton key={index} onClick={action} aria-label={label} color={color}>
```
Comment (MajaharZemoso) [inline src/components/molecules/Actions/index.tsx:22]: don't add index as a key

File: src/components/organisms/AddBooks/index.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        id: uuidv4(),
+        title: "",
+        author: "",
+        genre: "",
+        totalCopies: 0,
+    });
+
+    const navigate = useNavigate();
+
+    const returnBack = () => {
```
Comment (MajaharZemoso) [inline src/components/organisms/AddBooks/index.tsx:24]: add proper names to functions

File: src/components/organisms/AddBooks/index.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    return (
+        <CustomGrid>
+            <Grid item>
+                <CustomHeader headertitle={addBooksTitle} handleAdd={addBook} />
+            </Grid>
+            <Grid item>
+                <Text variants="subtitle1" text="Title" />
+                <CustomInput
+                    name="title"
```
Comment (MajaharZemoso) [inline src/components/organisms/AddBooks/index.tsx:67]: take strings from constants

FYI - this is for all files

File: src/components/organisms/AddBooks/styles.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Grid, styled, TextField } from "@mui/material";
+
+export const CustomGrid=styled(Grid)({
+    justifyContent:"center",
+    alignItems:"center",
+    display:"flex",
+    flexDirection:"column",
+    height:'100vh',
+    gap:"10px"
```
Comment (MajaharZemoso) [inline src/components/organisms/AddBooks/styles.tsx:9]: Don't use px instead you can use theme.spacing or rem

FYI - this comment is for all files

File: src/components/organisms/BooksTable/index.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            } catch (error) {
+                console.error("Error fetching books:", error);
+            }
+        }
+        getBooks();
+    }, []);
+
+   
+
+    const  getBook = (id: string) => {
```
Comment (MajaharZemoso) [inline src/components/organisms/BooksTable/index.tsx:38]: +1

File: src/components/organisms/BooksTable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        navigate('/add');
+    }
+
+    return (
+        <Box sx={{marginTop:'15px'}}>   
+            <Grid container>
+                <Grid item xs={12} >
+                    <CustomHeader headertitle={headerTitle} handleAdd={addBook} />
+                </Grid>
+                {/* <Grid item xs={1}>
```
Comment (MajaharZemoso) [inline src/components/organisms/BooksTable/index.tsx:78]: Remove commented code

File: src/components/organisms/DeleteBooks/index.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+                alert("Book not deleted");
+
+            }
+            navigate("/");
+        } catch (error) {
+            console.error("Error adding book:", error);
+            alert("Failed to add book");
+        }
+    };
+    const addBook = () => {
```
Comment (MajaharZemoso) [inline src/components/organisms/DeleteBooks/index.tsx:55]: +1

File: src/components/organisms/GetBooks/index.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+import Text from "../../atoms/Typography";
+import CustomButton from "../../atoms/Buttons";
+import { dataprops } from "../../../utils/interfaces";
+import { getBooksTitle } from "../../../utils/constants";
+
+const GetBook: React.FC = () => {
+    const navigate=useNavigate();
+    const {id}=useParams();
+
+    const returnBack = () => {
```
Comment (MajaharZemoso) [inline src/components/organisms/GetBooks/index.tsx:16]: +1

File: src/components/organisms/NotFoundPage/index.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Typography ,Box} from "@mui/material";
+import Text from "../../atoms/Typography"; 
+
+const NotFound: React.FC = () => {
+    // const navigate = useNavigate();
```
Comment (MajaharZemoso) [inline src/components/organisms/NotFoundPage/index.tsx:6]: +1

File: src/App.tsx (hunk 1 above)
Comment (SuduguTejasvi) [reply src/App.tsx]: Addressed it

File: src/components/StudentInformation.tsx
Comment (SuduguTejasvi) [reply src/components/StudentInformation.tsx:1]: Addressed it

File: src/components/molecules/Actions/index.tsx (hunk 2 above)
Comment (SuduguTejasvi) [reply src/components/molecules/Actions/index.tsx]: Addressed it

File: src/components/molecules/Actions/index.tsx (hunk 3 above)
Comment (SuduguTejasvi) [reply src/components/molecules/Actions/index.tsx]: Addressed it

File: src/components/organisms/AddBooks/index.tsx (hunk 4 above)
Comment (SuduguTejasvi) [reply src/components/organisms/AddBooks/index.tsx]: Addressed it

File: src/components/organisms/AddBooks/index.tsx (hunk 5 above)
Comment (SuduguTejasvi) [reply src/components/organisms/AddBooks/index.tsx:81]: Addressed it

File: src/components/organisms/AddBooks/styles.tsx (hunk 6 above)
Comment (SuduguTejasvi) [reply src/components/organisms/AddBooks/styles.tsx]: Addressed it

File: src/components/organisms/BooksTable/index.tsx (hunk 7 above)
Comment (SuduguTejasvi) [reply src/components/organisms/BooksTable/index.tsx]: Addressed it

File: src/components/organisms/BooksTable/index.tsx (hunk 8 above)
Comment (SuduguTejasvi) [reply src/components/organisms/BooksTable/index.tsx]: Addressed it

File: src/components/organisms/GetBooks/index.tsx (hunk 10 above)
Comment (SuduguTejasvi) [reply src/components/organisms/GetBooks/index.tsx]: Addressed it

File: src/components/organisms/NotFoundPage/index.tsx (hunk 11 above)
Comment (SuduguTejasvi) [reply src/components/organisms/NotFoundPage/index.tsx]: Addressed it

**Review by MajaharZemoso — APPROVED** [review]
